integr8ly/cluster-service
# add s3 bucket deletion via an action engine

add a new aws action engine for deleting s3 buckets associated
with a cluster.

this s3 action engine uses the aws tagging functionality to
filter only the s3 buckets that match the cluster id and
additional specified tags on the aws side instead of performing
filtering on the client side.

this commit also includes hack files to help with s3 manual
testing and future development.

verification:
- run 'make build/cli'
- run './hack/aws-s3-create-bucket.sh'
- run './cli cleanup cluster-service --region=us-east-1 --dry-run=false'
- ensure s3 buckets are removed for the cluster
